status-im/status-react
# chore: bump status-go with filter improvements

fixes issues noticed in lightClient using Waku-Filter protocol.

### Summary

Improved lightClient functionality especially fixing issues around Waku-Filter protocol usage and optimizing how filter subscriptions are managed and Filter ping occurs with service node. 
Still some optimizations and additional work is pending wrt light client reliability, but this PR brings Filter to a hopefully decent stage(without any major issues) before proceeding with further changes.

Note that if you are subscribed to status community, there would be a lot of filter subscriptions. This still needs optimization which is pending. Anyways, number of subscriptions would help test nwaku node as well to handle soo may :)

~~**Note that to reduce issues with lightClient(by not connecting to other desktop instances), this build only uses fleet nodes and does not do any discovery via peerExchange.**~~

Relevant status-go PR https://github.com/status-im/status-go/pull/4665

### Review notes


### Testing notes (To be updated)
Note to enable debug logs while testing.
- ~~Enable lightClient from Advanced Settings~~ @chaitanyaprem has updated PR so lightClient is Enabled by default
- Login normally, and make sure you got some peers connected  -- Advanced -> Peer Stats
- Run tests by being part of community , 1:1 chats, private group chats etc
- Disconnect from the internet, wait until the peer count goes to 0.  (Note that it takes some time for identifying disconnection - this needs fixing)
- Connect to the internet, You should reconnect to some peers in sometime. (Note that it takes some time for peers to be connected - this needs fixing)
- Test reception of messages on 1:1 chat, groups and community
- Leave it running and ensure no issues in send/receive messages

**In case of any issue please report along with debug logs via DM or as a comment in this PR itself.** 

#### Platforms


- Android
- iOS

#### Areas that maybe impacted


##### Functional

- 1-1 chats
- public chats
- group chats
- account recovery
- new account
- user profile updates
- networks
- fleet
- bootnodes

##### Non-functional

None


status: for dogfooding only
